public/components/map_container/maps_footer.tsx

```diff
@@ -4,17 +4,33 @@
  */
 
 import { EuiPanel } from '@elastic/eui';
-import { isEqual } from 'lodash';
-import React, { memo } from 'react';
-import { LngLat } from 'maplibre-gl';
+import React, { memo, useEffect, useState } from 'react';
+import { LngLat, MapEventType } from 'maplibre-gl';
+import { Map as Maplibre } from 'maplibre-gl';
 
 const coordinatesRoundOffDigit = 4;
 interface MapFooterProps {
-  coordinates?: LngLat;
+  map: Maplibre;
   zoom: number;
 }
 
-export const MapsFooter = memo(({ coordinates, zoom }: MapFooterProps) => {
+export const MapsFooter = memo(({ map, zoom }: MapFooterProps) => {
+  const [coordinates, setCoordinates] = useState<LngLat>();
+  useEffect(() => {
+    function onMouseMoveMap(e: MapEventType['mousemove']) {
+      setCoordinates(e.lngLat.wrap());
+    }
+
+    if (map) {
+      map.on('mousemove', onMouseMoveMap);
+    }
+    return () => {
+      if (map) {
+        map.off('mousemove', onMouseMoveMap);
+      }
+    };
+  }, []);
```

Comment:
Is it expected to only run once?

Reply:
Yes. We need to register these functions once per reder